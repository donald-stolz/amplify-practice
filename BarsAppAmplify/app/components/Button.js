// @flow
import React from 'react';
import {
  TouchableOpacity, View, Text, Dimensions, StyleSheet,
} from 'react-native';

// Types
type Props = {
  title: string,
  onPress: Function,
  style: {},
  textStyle?: {},
};

const Button = ({
  title, onPress, style, textStyle,
}: Props) => (
  <TouchableOpacity onPress={onPress}>
    <View style={[styles.buttonContainer, style]}>
      <Text style={[styles.buttonText, textStyle]}>{title}</Text>
    </View>
  </TouchableOpacity>
);

const { width } = Dimensions.get('window');

const styles = StyleSheet.create({
  buttonContainer: {
    backgroundColor: 'grey',
    marginBottom: 5,
    paddingLeft: 40,
    paddingRight: 40,
    paddingTop: 10,
    paddingBottom: 10,
    width: width * 0.8,
    borderRadius: 8,
  },
  buttonText: {
    color: 'white',
    fontSize: 16,
    textAlign: 'center',
    fontWeight: '900',
    letterSpacing: 2,
  },
});

Button.defaultProps = {
  textStyle: {},
};

export default Button;
