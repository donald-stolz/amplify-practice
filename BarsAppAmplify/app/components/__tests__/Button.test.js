import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../Button';

test('Button renders correctly', () => {
  const onPress = jest.fn();
  const tree = renderer.create(<Button title="Submit" onPress={onPress} />).toJSON();
  expect(tree).toMatchSnapshot();
});
