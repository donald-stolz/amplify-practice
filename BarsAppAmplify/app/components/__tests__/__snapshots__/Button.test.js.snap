// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "grey",
          "borderRadius": 8,
          "marginBottom": 5,
          "paddingBottom": 10,
          "paddingLeft": 40,
          "paddingRight": 40,
          "paddingTop": 10,
          "width": 600,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "900",
            "letterSpacing": 2,
            "textAlign": "center",
          },
          Object {},
        ]
      }
    >
      Submit
    </Text>
  </View>
</View>
`;
